Add behavioral subfeature data guideline

ddbeck opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->

#### Summary

This PR provides a general guideline for creating behavioral subfeatures.

<!-- ✍️ In a sentence or two, describe your changes. -->

#### Related issues

This continues a discussion started in https://github.com/mdn/browser-compat-data/pull/26605#issuecomment-2838278224, while https://github.com/mdn/browser-compat-data/pull/26780 covers the partial implementation part of this.

This was partly inspired by https://github.com/mdn/browser-compat-data/pull/17352, but is narrower in scope.

<!-- 🔨 If applicable, use "Fixes #XYZ" -->

<!-- ✅ After submitting, review the results of the "Checks" tab! -->

caugner reviewed (commented):
LGTM so far.

Like https://github.com/mdn/browser-compat-data/pull/26780 already includes, some more examples might help comprehension.

ddbeck commented:
@caugner I already mentioned 3 examples (`css.properties.gap.flex_context` and `css.properties.gap.grid_context`, and `html.global_attributes.inert.ignores_find_in_page`) and mention more specific guidelines for (e.g., `secure_context_required`). Do you have something in mind here?

caugner commented on docs/data-guidelines/index.md:
```diff
@@ -160,6 +160,27 @@ Do not set `deprecated` to `true` for features that are merely old or unpopular,
+
+You may create a behavioral subfeature for the following types of browser behaviors:
+
+- A contextual support condition this is not readily feature detectable, such as a CSS layout context or the effect of [an IDL extended attribute](https://dontcallmedom.github.io/webidlpedia/extended-attributes.html).
+
+  For example, if support for a CSS property is conditioned on the layout context where that property is used, create a subfeature of the property to represent the behavior (as demonstrated by `css.properties.gap.flex_context` and `css.properties.gap.grid_context`).
```
Here's what I had in mind:

```suggestion
- When support depends on a contextual condition that cannot be easily or accurately feature-detected.

  Examples:
  1. CSS layout context: The CSS property `gap` depends on the layout context, so there are behavioral subfeatures `flex_context` and `grid_context` to distinguish support (see: [css/properties/gap.json](https://github.com/mdn/browser-compat-data/blob/main/css/properties/gap.json)).
  2. Effects of [an IDL extended attribute](https://dontcallmedom.github.io/webidlpedia/extended-attributes.html) <!-- Add example -->
```

Same for the behavioral evolution.

ddbeck commented on docs/data-guidelines/index.md:
```diff
@@ -160,6 +160,41 @@ Do not set `deprecated` to `true` for features that are merely old or unpopular,
+
+You may create a behavioral subfeature for the following types of browser behaviors:
+
+- A contextual support condition this is not readily feature detectable.
```
```suggestion
- A contextual support condition that is not readily feature detectable.
```

Elchi3 reviewed (approved):
LGTM, thanks for working on this!

- Maybe it is stated somewhere already but I think behavioral subfeatures should always have a description.
- I have not yet looked at data but my feeling is that often these features don't age well because very often they are introduced as part of documenting browser release notes. Not sure we need dedicated guidance (or irrelevance criteria) right now, but maybe something to keep in mind for later.
- If we see patterns of behavioral subfeatures, we should introduce dedicated guidelines (covered by the second paragraph you added), so that the data becomes more consistent.

caugner reviewed (approved):
LGTM.

Let's:

- add a note about these guidelines being new
- add this to the `docs/index.md`

ddbeck commented:
This came up again on today's BCD call. I've made a few small changes:

- Added text requiring a `description`.
- Added links to all more-specific behavioral subfeature guidelines.
- Added "feedback welcome" text at the top of the guideline. In today's call, we talked about adopting the guideline with the understanding it's sort of provisional. We talked about removing this text in 6 months (assuming the guideline survives that long).
- Added this guideline to docs index.

@caugner and @Elchi3, I'd welcome re-review here. Thank you!
